homeassistant/components/amcrest/__init__.py

```diff
@@ -71,70 +76,110 @@
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_AUTHENTICATION, default=HTTP_BASIC_AUTHENTICATION):
+            vol.All(vol.In(AUTHENTICATION_LIST)),
+        vol.Optional(CONF_RESOLUTION, default=DEFAULT_RESOLUTION):
+            vol.All(vol.In(RESOLUTION_LIST)),
+        vol.Optional(CONF_STREAM_SOURCE, default=DEFAULT_STREAM_SOURCE):
+            vol.All(vol.In(STREAM_SOURCE_LIST)),
+        vol.Optional(CONF_FFMPEG_ARGUMENTS, default=DEFAULT_ARGUMENTS):
+            cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL):
+            cv.time_period,
+        vol.Optional(CONF_BINARY_SENSORS):
+            vol.All(cv.ensure_list_csv, [vol.In(BINARY_SENSORS)]),
+        vol.Optional(CONF_SENSORS):
+            vol.All(cv.ensure_list_csv, [vol.In(SENSORS)],
+                    _deprecated_sensor_values),
+        vol.Optional(CONF_SWITCHES):
+            vol.All(cv.ensure_list_csv, [vol.In(SWITCHES)]),
+    }),
+    _deprecated_switches
+)
 
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.All(cv.ensure_list, [AMCREST_SCHEMA], _has_unique_names)
 }, extra=vol.ALLOW_EXTRA)
 
+SERVICE_ENABLE_RECORDING = 'enable_recording'
+SERVICE_DISABLE_RECORDING = 'disable_recording'
+SERVICE_ENABLE_AUDIO = 'enable_audio'
+SERVICE_DISABLE_AUDIO = 'disable_audio'
+SERVICE_ENABLE_MOTION_RECORDING = 'enable_motion_recording'
+SERVICE_DISABLE_MOTION_RECORDING = 'disable_motion_recording'
+SERVICE_GOTO_PRESET = 'goto_preset'
+SERVICE_SET_COLOR_BW = 'set_color_bw'
+SERVICE_START_TOUR = 'start_tour'
+SERVICE_STOP_TOUR = 'stop_tour'
+
+ATTR_PRESET = 'preset'
+ATTR_COLOR_BW = 'color_bw'
+
+CBW_COLOR = 'color'
+CBW_AUTO = 'auto'
+CBW_BW = 'bw'
+CBW = [CBW_COLOR, CBW_AUTO, CBW_BW]
+
+SERVICE_GOTO_PRESET_SCHEMA = CAMERA_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_PRESET): vol.All(vol.Coerce(int), vol.Range(min=1)),
+})
+SERVICE_SET_COLOR_BW_SCHEMA = CAMERA_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_COLOR_BW): vol.In(CBW),
+})
+
 
 def setup(hass, config):
     """Set up the Amcrest IP Camera component."""
     from amcrest import AmcrestCamera, AmcrestError
 
-    hass.data.setdefault(DATA_AMCREST, {})
-    amcrest_cams = config[DOMAIN]
+    hass.data.setdefault(DATA_AMCREST, {'devices': {}, 'cameras': []})
+    devices = config[DOMAIN]
 
-    for device in amcrest_cams:
+    for device in devices:
         name = device[CONF_NAME]
         username = device[CONF_USERNAME]
         password = device[CONF_PASSWORD]
 
         try:
-            camera = AmcrestCamera(device[CONF_HOST],
-                                   device[CONF_PORT],
-                                   username,
-                                   password).camera
+            api = AmcrestCamera(device[CONF_HOST],
+                                device[CONF_PORT],
+                                username,
+                                password).camera
             # pylint: disable=pointless-statement
```

Comment:
Please add a comment why this call is not pointless

Reply:
Yes, I considered doing that. (This component, and the pypi.org package it uses, have been around for a while. I'm pretty late to the game. Trying not to solve all the world's problems in one PR. 😄 ) But I can certainly do that.